File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/MyPackage/RandomProvider.cpp
#include "RandomProvider.h"
#include "IMyInterface.h"

namespace MyPackage
{
  RandomProvider::RandomProvider()
  {
  }
  int RandomProvider::GenerateInt(int offset)
  {
    return this->dependency_GetNumber->GenerateInt(offset);
  }
  int RandomProvider::MyFactor()
  {
    return this->dependency_GetNumber->MyFactor();
  }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/MyPackage/RandomProvider.cs
/// <filename>
///     RandomProvider.cs
/// </filename>
using MyPackage;

namespace MyPackage
{
    public class RandomProvider : MyPackage.IMyInterface
    {

        public RandomProvider()
        {
        }

        public virtual int GenerateInt(int offset)
        {
            return this.dependency_GetNumber.GenerateInt(offset);
        }

        public virtual int MyFactor()
        {
            return this.dependency_GetNumber.MyFactor();
        }

        public MyPackage.IMyInterface dependency_GetNumber;
    }
}
